### ISSUE TITLE
Add the Option to skip adding contacts for Integrated Addresses

### ISSUE THREAD

**Issue by BenjaminMoe** [turtlecoin/turtlecoin-mobile-wallet#152]
**Describe the bug**

This is an issue that arises from working with integrated addresses when working with Turtlepay. In terms of ideal functionality I would like to have the shop name, amount and integrated address in the QR code, so the person buying the thing can open their phone, scan the QR code, and then simply be asked to confirm the transaction or not.

This works for the first time an integrated address is used. But a new contact will be created for that integrated address with that name. So when you try to use Turtlepay a second time for the same shop, a new integrated address will be generated for the same name. and TonChan will prompt the 

**To Reproduce**

My test environment is over at https://trtl.dashgl.com/. I have a single page with a simple model. Click on "Checkout" and the server will send a request to turtlepay to get a payment id, and then provide a QR to the use to scan. If you can the QRcode once, Tonchan will simply ask to confirm the payment. If you scan the QR code again, TonChan will prompt for a new name to be entered for a new contact before asking to confirm the transaction.

As a side note, the repository for this testing is here: https://github.com/BenjaminMoe/TrtlCheckout if you want to test with your own wallet. 

**Expected behavior**

Integrated addresses probably shouldn't be added as contacts. My expectation is that if you have a friend or family member and you have a fixed address then it makes sense to store them as a contact. Or if you have a shop, it would make sense to store a fixed address. For something like a checkout using an integrated address, it's a one-time transaction that's going to be unique every time. So adding a new contact every time seems like it would add a lot of clutter.

For integrated addresses my suggestion would be to skip contact registration entirely. Hopefully the name still shows up in the transaction history, but a contact is not needed. 

**Screenshots**

First time, this is what I would like to see everytime, simply a transaction to checkout.
![Screenshot_20210622-233449_TonChan](https://user-images.githubusercontent.com/86194145/122947543-33691c80-d3b5-11eb-9148-007ecbfe187c.png)

For the second time on, there will be a name conflict and 
![Screenshot_20210622-234117_TonChan-576x1024](https://user-images.githubusercontent.com/86194145/122947626-42e86580-d3b5-11eb-9295-0c8e9192dd4f.png)

**Environment**
TonChan v1.2.3 running LineageOS. 

**Additional context**

The context is to provide customers with a QR code they can scan and make a payment, and then verify that payment. The idea is to keep the keys in the hands of the customer and not have an environment where the vendor is holding onto private keys for their customers.

**Comment by zpalmtree** [comment]
Makes sense. I'm not sure if a user would *never* want to save an integrated address as a contact, some exchanges will keep using the same address+pid pair, but you are probably right that in most cases saving is not desired.

I guess it would be best to put a 'save to contacts' option somewhere.. maybe on the 'New payee' screen would work.
If we also then displayed that as a checkbox on the review screen, and defaulted to false when a QR with an integrated address is given, that should work.

I thought about making the name field optional if they don't save to contacts, but that would make toggling between saving and not saving annoying to implement.

**Comment by BenjaminMoe** [comment]
Or maybe if this is defined by the person receiving the funds? Maybe adding the option to "skipContacts=true' on the turtlecoin:// query string, might be an simpler option that doesn't require the behavior to be specified by the end-user. 

Another possibility is that I could be using the TurtlePay API wrong. On the TurtlePay '[Getting Started](https://turtlepay.io/start)' page, it says:

```Note: Payment requests are only valid for approximately 30 minutes (exactly 60 blocks). So it's important that if your user doesn't complete a payment that a new request is completed if they attempt to pay you again.```

So my impression is that to keep track of unique transactions that I would need to send a new request to the TurtlePay API with a new payment ID to be able to track if the payment has been made or not.

Another possibility that I'm not sure how feasible would be to poll my own wallet for new transactions. That way I could make a QR code with a static address, an amount, and a payment id, and then be able to use the payment id to confirm the transaction. That would mean that only one new contact would be added for a single address. 

Right now I'm a little confused about what kind of magic is behind an integrated address. Is it required for something like TurtlePay to work? And if I need to create a new integrated address each time. In this case for a proof of concept, I have a 3d model. A new payment id is issued, and then i can use that payment id to track if that user has paid to allow the download or not. So I don't think I can use the same integrated address over and over as it wouldn't give me context of who made the payment.

Unless the other alternative would be to generate a single integrated address for each new user. But I'm trying to make a shop that's anonymous. I shouldn't need to require a login or track the user with cookies, only check to see if a specific amount was paid using a specific payment id.
